File: C/18_Stack/BOJ_01_10828_스택.c
#include <stdio.h>
#include <stdlib.h>
int stack[10000];
int idx = -1;
void push(int value) {
	stack[++idx] = value;
}
int pop() {
	if (idx < 0) return -1;
	else return stack[idx--];
}
int size() {
	return idx + 1;
}
int empty() {
	if (idx < 0) return 1;
	else return 0;
}
int top() {
	if (idx == -1) {
		return -1;
	}
	else {
		return stack[idx];
	}
}
main() {
	int n, x;
	int cnt = 0;
	char arr[6];
	scanf("%d", &n);
	int* result = (int*)malloc(sizeof(int) * n);
	for (int i = 0; i < n; i++) {
		scanf("%s", arr);
		if (arr[0] == 'p') {
			if (arr[1] == 'o') result[cnt++] = pop();
			else {
				scanf("%d", &x);
				push(x);
			}
		}
		else if (arr[0] == 't') result[cnt++] = top();
		else if (arr[0] == 'e') result[cnt++] = empty();
		else result[cnt++] = size();
	}
	for (int i = 0; i < cnt; i++) printf("%d\n", result[i]);
	free(result);
}

File: JAVA/18_Stack/BOJ_01_10828_스택.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static int[] stack;
    public static int idx = -1;

    public static void push(int item) {
        stack[++idx] = item;
    }

    public static int pop() {
        if (idx < 0) {
            return -1;
        } else {
            return stack[idx--];
        }
    }

    public static int size() {
        return idx + 1;
    }

    public static int empty() {
        if (idx < 0) {
            return 1;
        } else {
            return 0;
        }
    }

    public static int top() {
        if (idx < 0) {
            return -1;
        } else {
            return stack[idx];
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());
        stack = new int[n];
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            String str = st.nextToken();
            if (str.contains("push")) push(Integer.parseInt(st.nextToken()));
            else if (str.contains("pop")) sb.append(pop()).append("\n");
            else if (str.contains("top")) sb.append(top()).append("\n");
            else if (str.contains("empty")) sb.append(empty()).append("\n");
            else if (str.contains("size")) sb.append(size()).append("\n");
        }
        System.out.println(sb);
    }
}